packages/vm/src/eei/eei.ts

```diff
@@ -94,6 +94,8 @@ export class EEI extends VmState implements EEIInterface {
   }
 
   public copy() {
-    return new EEI(this._stateManager.copy(), this._common.copy(), this._blockchain.copy())
+    const common = this._common.copy()
+    common.setHardfork(this._common.hardfork())
```

Comment:
Valid solution here, but I guess this hardfork setting should be functionality directly integrated into the `Common` `copy()` function? 🤔

Reply:
I wasn't sure on this one.  It might warrant further research on whether or not there is other "state" inside of `common` that needs to be copied along with the current `hardfork`.  That is the most impactful one (as identified by our failing tests) but I'm not sure if there's more to be thought through here.

Reply:
This got me a bit curious and a had another look, since on second thought it seemed extremely unlikely to me that our libraries would work at all if in the former setup `Common.copy()` wouldn't preserve the hardfork state. Tested this out and it turns out that it *does*, since it does copy property values as well in `Common.copy()`.

You can test out the following code:

```typescript
import { Common, Chain, Hardfork } from '@ethereumjs/common'
const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Shanghai, eips: [4844] })
const common2 = common.copy()
common2.hardfork() // 'shanghai'
```

So seems we can delete these additional lines here again, the fixes seem to come from some other code changes in this PR. 🙂

EIPs are actually preserved as well, `common2.eips() // [ 4844 ]`.

Reply:
Ah, okay. That makes sense. I wasn't sure if the EEI was impacted or not. We certainly had a bug where copying VM resulted in separate `common` instances being created for the internal evm and EEI but maybe that was the limit of the issue.